Improve description of iso_year and iso_week

jrlawhorne opened the pull request.

jrlawhorne committed:
Improve description of iso_year and iso_week

mgroves commented:
Thanks for the contribution @jrlawhorne :) I've added a couple reviewers from the docs team, so this has a little more visibility.

RichardSmedley reviewed (approved):
LGTM - thanks for the contribution.  :+1:

simon-dew committed:
Minor corrections.

Corrected a typo ("a an").
We tend to use underscore rather than caps for emphasis, so `_not_` rather than `NOT`, to avoid confusion with keywords.
Also clarify that the first `iso_week` is numbered 1.

simon-dew reviewed (approved):
👍 LGTM — made some minor corrections
